### ISSUE TITLE
Make CTR-DRBG, NIST-KW, CMAC work with Cipher light

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#8061]
- Change those modules from "depends on `CIPHER_C`" to "auto-enables `CIPHER_LIGHT` in `build-info.h`".
- Re-enable them in `all.sh` component `test_cipher_light_only`
- For CTR-DRBG, remove the non-cipher implementation that was temporarily kept in #8054

**Depends on:** #8058, #8054

**Comment by gilles-peskine-arm** [comment]
Based on the [current plans for cipher light](https://github.com/Mbed-TLS/mbedtls/pull/8057), this is out of scope.

**Comment by mpg** [comment]
Agreed - except for the part about CTR-DRBG, which I've just moved to #8060. 

As a result, closing this issue as out of scope with respect to [the current goals](https://github.com/Mbed-TLS/mbedtls/issues/8337).
